libvast/src/system/spawn_arguments.cpp

```diff
@@ -35,10 +38,24 @@ normalized_and_validated(std::vector<std::string>::const_iterator begin,
                          std::vector<std::string>::const_iterator end) {
   if (begin == end)
     return caf::make_error(ec::syntax_error, "no query expression given");
-  if (auto e = to<expression>(caf::join(begin, end, " ")))
+  auto str = detail::join(begin, end, " ");
+  // TODO: improve error handling. Right now, we silently try to parse the
+  // query expression as Sigma rule. If it fails, we only print a debug log
+  // entry and move on to parsing the input as VAST expression.
```

Comment:
Please write a follow-up story for this, it seems rather important.

Reply:
I think the behavior is actually correct here. The question is: maybe we should only fail if we have valid YAML but not a valid Sigma rule? Since this is the auto-detection part: we must go to the next form input after YAML parsing failed *without* terminating VAST.

Reply:
Why not require a `--sigma` flag and support either instead of trying both?

Reply:
This might be additive to lock in a specific syntax, but by default I like the UX that VAST tries all known options.

Reply:
I wrote a story for revisiting this as well.